### DISCUSSION TITLE
2024.10.0

### DISCUSSION THREAD

**Discussion by AndrewBastin** [hoppscotch/hoppscotch#4499] [Announcements]
This releases introduces support for HTTP Proxy and CA certificate support in the desktop app, Digest auth flow as an auth mechanism and compatibility improvements for selfhosted containers particularly hosted in rootless (Podman, Rootless Docker) or hardened container runtimes (OpenShift). 

## What's Changed
* HSB-489 feat: hoppscotch-backend dependency update by @mirarifhasan in https://github.com/hoppscotch/hoppscotch/pull/4426
* HSB-494 fix: downgrade eslint related package by @mirarifhasan in https://github.com/hoppscotch/hoppscotch/pull/4445
* fix: prevent keypress actions in the loading state by @xLansang in https://github.com/hoppscotch/hoppscotch/pull/4462
* fix: resolve empty state image correctly in profile page by @jamesgeorge007 in https://github.com/hoppscotch/hoppscotch/pull/4456
* feat(common): introducing ability to set custom collection import size limit by @JoelJacobStephen in https://github.com/hoppscotch/hoppscotch/pull/4440
* feat: add the ability to configure query params encoding for requests by @Pranay-Pandey in https://github.com/hoppscotch/hoppscotch/pull/4412
* chore(sh-admin): bump dependencies by @JoelJacobStephen in https://github.com/hoppscotch/hoppscotch/pull/4433
* chore(agent): major version dependency bump by @CuriousCorrelation in https://github.com/hoppscotch/hoppscotch/pull/4446
* chore(desktop): major version dependency bumps by @CuriousCorrelation in https://github.com/hoppscotch/hoppscotch/pull/4443
* chore: bump dependencies by @jamesgeorge007 in https://github.com/hoppscotch/hoppscotch/pull/4444
* feat: `openssl` based `hoppscotch-relay` for request forwarding by @CuriousCorrelation in https://github.com/hoppscotch/hoppscotch/pull/4442
* feat: portable version of `hoppscotch-agent` by @CuriousCorrelation in https://github.com/hoppscotch/hoppscotch/pull/4468
* feat(ci): portable and installer build workflow for `hoppscotch-agent` by @CuriousCorrelation in https://github.com/hoppscotch/hoppscotch/pull/4472
* feat: `hoppscotch-relay` for request forwarding by @CuriousCorrelation in https://github.com/hoppscotch/hoppscotch/pull/4471
* chore: environment highlight colour schema update by @nivedin in https://github.com/hoppscotch/hoppscotch/pull/4464
* fix(i18n/es): fixed translation errors and translated missing keys by @sawa-ko in https://github.com/hoppscotch/hoppscotch/pull/4455
* feat(i18n/cn): update CN translation by @subframe7536 in https://github.com/hoppscotch/hoppscotch/pull/4457
* chore: improve `German` translation by @RaHehl in https://github.com/hoppscotch/hoppscotch/pull/4461
* fix: codegen breaking in example tab and minor UI updates by @nivedin in https://github.com/hoppscotch/hoppscotch/pull/4478
* fix: postman env variables are to be imported as secrets by @shobanrajm in https://github.com/hoppscotch/hoppscotch/pull/4474
* fix: trailing comma makes json invalid by @anwarulislam in https://github.com/hoppscotch/hoppscotch/pull/4416
* feat: add button to export test results as json by @Pranay-Pandey in https://github.com/hoppscotch/hoppscotch/pull/4419
* HSB-495 feat: user workspace stats api added by @mirarifhasan in https://github.com/hoppscotch/hoppscotch/pull/4467
* feat: ability for aio in sub-path access configurations to specify alternate ports by @AndrewBastin in https://github.com/hoppscotch/hoppscotch/pull/4480
* feat: add support for `Digest` authorization by @anwarulislam in https://github.com/hoppscotch/hoppscotch/pull/4339
* feat(desktop): add CA cert and HTTP proxy support for native interceptor by @CuriousCorrelation in https://github.com/hoppscotch/hoppscotch/pull/4491
* fix: scrollbar appearing on digest auth input fields and other improvements by @anwarulislam in https://github.com/hoppscotch/hoppscotch/pull/4492
* fix: process headers correctly in Digest Auth and other updates by @jamesgeorge007 in https://github.com/hoppscotch/hoppscotch/pull/4494
* chore(agent): bump version to `0.1.2` by @CuriousCorrelation in https://github.com/hoppscotch/hoppscotch/pull/4497
* chore: introduce `supportsDigestAuth` field at the interceptor service level by @jamesgeorge007 in https://github.com/hoppscotch/hoppscotch/pull/4498

## New Contributors
* @xLansang made their first contribution in https://github.com/hoppscotch/hoppscotch/pull/4462
* @Pranay-Pandey made their first contribution in https://github.com/hoppscotch/hoppscotch/pull/4412
* @subframe7536 made their first contribution in https://github.com/hoppscotch/hoppscotch/pull/4457
* @RaHehl made their first contribution in https://github.com/hoppscotch/hoppscotch/pull/4461
* @shobanrajm made their first contribution in https://github.com/hoppscotch/hoppscotch/pull/4474

**Full Changelog**: https://github.com/hoppscotch/hoppscotch/compare/2024.9.3...2024.10.0

<hr /><em>This discussion was created from the release <a href='https://github.com/hoppscotch/hoppscotch/releases/tag/2024.10.0'>2024.10.0</a>.</em>

**Comment by Leon-Luu** [comment]
Hi @AndrewBastin 
We got errors on the errors on "Setup pnpm" step for the "Node.js CI" workflow with the latest release.

![image](https://github.com/user-attachments/assets/230f092f-376e-455f-acb9-77fb7d1abe8c)

**Reply by jamesgeorge007** [reply]
Hi, this is fixed in #4514.

**Comment by Leon-Luu** [comment]
There is flaky test related to packages/hoppscotch-cli where we could see this issues (2 out of 3 times in pipeline)

![image](https://github.com/user-attachments/assets/d2dd30c1-822f-400d-9fd3-0f428a3f1c1c)

**Reply by Leon-Luu** [reply]
In windows machine I got 100% errors :

 ✓ src/__tests__/unit/getters.spec.ts  (21 tests) 53ms
│ (node:39192) [DEP0147] DeprecationWarning: In future versions of Node.js, fs.rmdir(path, { recursive: true }) will be removed. Use fs.rm(path, { recursive: true }) instead
│ (Use `node --trace-deprecation ...` to show where the warning was created)
│  ❯ src/__tests__/e2e/commands/test.spec.ts  (45 tests | 5 failed) 99961ms
│    × hopp test [options] <file_path_or_id> > Test `hopp test <file_path_or_id> --env <file_path_or_id>` command: > AWS Signature Authorization type > Successfully translates the authorization information to headers/query params and sends …  
│      → expected Error: Command failed: node F:\develop\te… { …(4) } to be null
│    × hopp test [options] <file_path_or_id> > Test `hopp test <file_path_or_id> --env <file_path_or_id>` command: > Digest Authorization type > Successfully translates the authorization information to headers/query params and sends it alon…  
│      → expected Error: Command failed: node F:\develop\te… { …(4) } to be null
│    × hopp test [options] <file_path_or_id> > Test`hopp test <file_path_or_id> --env <file_path_or_id> --reporter-junit [path] > Report export fails with the code `REPORT_EXPORT_FAILED` while encountering an error during path creation        
│      → expected '' to be 'REPORT_EXPORT_FAILED' // Object.is equality
│    × hopp test [options] <file_path_or_id> > Test`hopp test <file_path_or_id> --env <file_path_or_id> --reporter-junit [path] > Generates a JUnit report at the default path
│      → Snapshot `hopp test [options] <file_path_or_id> > Test`hopp test <file_path_or_id> --env <file_path_or_id> --reporter-junit [path] > Generates a JUnit report at the default path 1` mismatched
│    × hopp test [options] <file_path_or_id> > Test`hopp test <file_path_or_id> --env <file_path_or_id> --reporter-junit [path] > Generates a JUnit report at the specified path
│      → Snapshot `hopp test [options] <file_path_or_id> > Test`hopp test <file_path_or_id> --env <file_path_or_id> --reporter-junit [path] > Generates a JUnit report at the specified path 1` mismatched
│ ⎯⎯⎯⎯⎯⎯⎯ Failed Tests 5 ⎯⎯⎯⎯⎯⎯⎯
│  FAIL  src/__tests__/e2e/commands/test.spec.ts > hopp test [options] <file_path_or_id> > Test `hopp test <file_path_or_id> --env <file_path_or_id>` command: > AWS Signature Authorization type > Successfully translates the authorization in…  
│ AssertionError: expected Error: Command failed: node F:\develop\te… { …(4) } to be null
│  ❯ src/__tests__/e2e/commands/test.spec.ts:418:23
│     416|         const { error } = await runCLI(args, { env });
│     417|
│     418|         expect(error).toBeNull();
│        |                       ^
│     419|       });
│     420|     });
│ ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[1/5]⎯
│  FAIL  src/__tests__/e2e/commands/test.spec.ts > hopp test [options] <file_path_or_id> > Test `hopp test <file_path_or_id> --env <file_path_or_id>` command: > Digest Authorization type > Successfully translates the authorization informati…  
│ AssertionError: expected Error: Command failed: node F:\develop\te… { …(4) } to be null
│  ❯ src/__tests__/e2e/commands/test.spec.ts:436:23
│     434|         const { error } = await runCLI(args);
│     435|
│     436|         expect(error).toBeNull();
│        |                       ^
│     437|       });
│     438|     });
│ ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[2/5]⎯
│  FAIL  src/__tests__/e2e/commands/test.spec.ts > hopp test [options] <file_path_or_id> > Test`hopp test <file_path_or_id> --env <file_path_or_id> --reporter-junit [path] > Report export fails with the code `REPORT_EXPORT_FAILED` while enc…  
│ AssertionError: expected '' to be 'REPORT_EXPORT_FAILED' // Object.is equality
│ - Expected
│ + Received
│ - REPORT_EXPORT_FAILED
│  ❯ src/__tests__/e2e/commands/test.spec.ts:662:19
│     660|
│     661|       const out = getErrorCode(stderr);
│     662|       expect(out).toBe<HoppErrorCode>("REPORT_EXPORT_FAILED");
│        |                   ^
│     663|
│     664|       expect(stdout).not.toContain(
│ ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[3/5]⎯
│  FAIL  src/__tests__/e2e/commands/test.spec.ts > hopp test [options] <file_path_or_id> > Test`hopp test <file_path_or_id> --env <file_path_or_id> --reporter-junit [path] > Generates a JUnit report at the default path
│ Error: Snapshot `hopp test [options] <file_path_or_id> > Test`hopp test <file_path_or_id> --env <file_path_or_id> --reporter-junit [path] > Generates a JUnit report at the default path 1` mismatched
│ - Expected
│ + Received
│ @@ -1,10 +1,10 @@
│   "<?xml version="1.0" encoding="UTF-8"?>
│   <testsuites tests="76" failures="2" errors="66" time="time">
│     <testsuite name="test-junit-report-export/request-level-errors/invalid-url" time="time" timestamp="timestamp" tests="22" failures="0" errors="22">
│       <system-err><![CDATA[
│ -       REQUEST_ERROR - TypeError: Invalid URL]]></system-err>
│ +       REQUEST_ERROR - TypeError [ERR_INVALID_URL]: Invalid URL]]></system-err>
│       <testcase name="`toBeLevelxxx()` error scenarios - Expected 200-level status but could not parse value 'foo'" classname="test-junit-report-export/request-level-errors/invalid-url">
│         <error message="Expected 200-level status but could not parse value 'foo'"/>
│       </testcase>
│       <testcase name="`toBeLevelxxx()` error scenarios - Expected 200-level status but could not parse value 'foo'" classname="test-junit-report-export/request-level-errors/invalid-url">
│         <error message="Expected 200-level status but could not parse value 'foo'"/>
│ @@ -70,11 +70,11 @@
│         <error message="Argument for toInclude should not be undefined"/>
│       </testcase>
│     </testsuite>
│     <testsuite name="test-junit-report-export/request-level-errors/test-script-reference-error" time="time" timestamp="timestamp" tests="0" failures="0" errors="0">
│       <system-err><![CDATA[
│ -       REQUEST_ERROR - TypeError: Invalid URL
│ +       REQUEST_ERROR - TypeError [ERR_INVALID_URL]: Invalid URL
│         TEST_SCRIPT_ERROR - Script execution failed: ReferenceError: status is not defined]]></system-err>
│     </testsuite>
│     <testsuite name="test-junit-report-export/request-level-errors/non-existent-env-var" time="time" timestamp="timestamp" tests="22" failures="0" errors="22">
│       <system-err><![CDATA[
│         PARSING_ERROR - {
│  ❯ src/__tests__/e2e/commands/test.spec.ts:695:55
│   Snapshots  2 failed
│  Test Files  1 failed | 2 passed (3)
│       Tests  5 failed | 65 passed (70)
│    Start at  11:10:10
│    Duration  102.40s (transform 1.61s, setup 1.37s, collect 3.82s, tests 100.10s, environment 5ms, prepare 1.83s)
│     693|         .toString();
│     694|
│     695|       expect(replaceDynamicValuesInStr(fileContents)).toMatchSnapshot(…
│        |                                                       ^
│     696|     });
│     697|
│ ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[4/5]⎯
│  FAIL  src/__tests__/e2e/commands/test.spec.ts > hopp test [options] <file_path_or_id> > Test`hopp test <file_path_or_id> --env <file_path_or_id> --reporter-junit [path] > Generates a JUnit report at the specified path
│ Error: Snapshot `hopp test [options] <file_path_or_id> > Test`hopp test <file_path_or_id> --env <file_path_or_id> --reporter-junit [path] > Generates a JUnit report at the specified path 1` mismatched
│ - Expected
│ + Received
│ @@ -1,10 +1,10 @@
│   "<?xml version="1.0" encoding="UTF-8"?>
│   <testsuites tests="76" failures="2" errors="66" time="time">
│     <testsuite name="test-junit-report-export/request-level-errors/invalid-url" time="time" timestamp="timestamp" tests="22" failures="0" errors="22">
│       <system-err><![CDATA[
│ -       REQUEST_ERROR - TypeError: Invalid URL]]></system-err>
│ +       REQUEST_ERROR - TypeError [ERR_INVALID_URL]: Invalid URL]]></system-err>
│       <testcase name="`toBeLevelxxx()` error scenarios - Expected 200-level status but could not parse value 'foo'" classname="test-junit-report-export/request-level-errors/invalid-url">
│         <error message="Expected 200-level status but could not parse value 'foo'"/>
│       </testcase>
│       <testcase name="`toBeLevelxxx()` error scenarios - Expected 200-level status but could not parse value 'foo'" classname="test-junit-report-export/request-level-errors/invalid-url">
│         <error message="Expected 200-level status but could not parse value 'foo'"/>
│ @@ -70,11 +70,11 @@
│         <error message="Argument for toInclude should not be undefined"/>
│       </testcase>
│     </testsuite>
│     <testsuite name="test-junit-report-export/request-level-errors/test-script-reference-error" time="time" timestamp="timestamp" tests="0" failures="0" errors="0">
│       <system-err><![CDATA[
│ -       REQUEST_ERROR - TypeError: Invalid URL
│ +       REQUEST_ERROR - TypeError [ERR_INVALID_URL]: Invalid URL
│         TEST_SCRIPT_ERROR - Script execution failed: ReferenceError: status is not defined]]></system-err>
│     </testsuite>
│     <testsuite name="test-junit-report-export/request-level-errors/non-existent-env-var" time="time" timestamp="timestamp" tests="22" failures="0" errors="22">
│       <system-err><![CDATA[
│         PARSING_ERROR - {
│  ❯ src/__tests__/e2e/commands/test.spec.ts:724:55
│     722|         .toString();
│     723|
│     724|       expect(replaceDynamicValuesInStr(fileContents)).toMatchSnapshot(…
│        |                                                       ^
│     725|     });
│     726|
│ ⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯⎯[5/5]⎯
│  ELIFECYCLE  Test failed. See above for more details.

**Reply by jamesgeorge007** [reply]
> There is flaky test related to packages/hoppscotch-cli where we could see this issues (2 out of 3 times in pipeline)

This is a known issue; have specified a higher timeout at the test suite level via https://github.com/hoppscotch/hoppscotch/commit/3fc43cc9b96d519e81fa5be7a606dd1a0e617f73.

Regarding the failed CLI tests on Windows, can you create an issue with relevant information about the environment (OS build no., Node.js version ...), etc to reproduce?
Also, have you encountered any issues while using the CLI on the Windows platform?

**Reply by Leon-Luu** [reply]
Thanks for sharing @jamesgeorge007. CLI in windows works as expected on usage.
Issue [4520 ](https://github.com/hoppscotch/hoppscotch/issues/4520)is created
